airflow/providers/trino/hooks/trino.py

```diff
@@ -251,3 +251,33 @@ def test_connection(self):
             message = str(e)
 
         return status, message
+
+    @staticmethod
+    def _generate_insert_sql(table, values, target_fields, replace, **kwargs):
+        """
+        Static helper method that generates the INSERT SQL statement.
+        The REPLACE variant is specific to MySQL syntax.
+
+        :param table: Name of the target table
+        :param values: The row to insert into the table
+        :param target_fields: The names of the columns to fill in the table
+        :param replace: Whether to replace instead of insert
+        :return: The generated INSERT or REPLACE SQL statement
+        :rtype: str
+        """
+        placeholders = [
+            "?",
```

Comment:
This is a lot of code duplication for changing just one char.
Sqlite provider also duplicate this function just to change this one char...
Maybe we can think of a better way to handle this?

Reply:
We could add a `placeholder` parameter to `DbApiHook._generate_insert_sql()` function, which would be passed from `**kwargs` in `DbApiHook.insert_rows()`. The parameter could have a default value of "%s".

Reply:
Yep. This is better. Especially that now we have common.sql provider and such change can be released together (and we can add `common-sql-provider>1.2.0` as limit :D.

One of the benefits of having the `common.sql` package which is independent from airlfow.

Reply:
Just committed the discussed functionality. Should I also modify the SQLite implementation in this PR?

Reply:
Sure you can (just make sure commit message and PR title/description) reflect this.

BTW can you check also with PrestoHook?
Usualy issues with Trino apply also to presto

Reply:
Can you also take a look at `_generate_insert_sql` in `PostgresHook`?
I'm not sure there is action there but just to confirm

Reply:
Yep, just had a look and it has Postgres-specific SQL generation, and the placeholder used is `"%s"` which is [correct for psycopg2](https://www.psycopg.org/docs/usage.html#passing-parameters-to-sql-queries)

Reply:
thanks
Note to the comment in https://github.com/apache/airflow/pull/25939#discussion_r953958410
By adding this change the effected providers are depended on a new release of common-sql
so we should edit the provider.yml :
https://github.com/apache/airflow/blob/e996a88c7b19a1d30c529f5dd126d0a8871f5ce0/airflow/providers/trino/provider.yaml#L40

and set the dependency as `>=1.2.0`

Otherwise we are risking users installing the new version of Trino provider with older version of common.sql provider which will result in code not working

(Same goes for the other effected providers)

Reply:
When should we expect 1.2.0 to be released?

Reply:
it will be the next release. providers are released together so it's fine